### PR TITLE
pict-rs: 0.3.3 -> 0.4.0

### PR DESCRIPTION
**Pull request by adisbladis**
###### Description of changes

This upgrades pict-rs to 0.4.0 while keeping 0.3 in place for `stateVersion` <= 23.05.
The pict-rs developer @asonix commented on how to use the auto-migration feature of it [here](https://github.com/NixOS/nixpkgs/pull/243218#issuecomment-1634530059) but I don't think we should do that.

Commonly in NixOS when a database configuration format changes we don't migrate users data. We rely on `stateVersion` to ensure that a version compatible with the data is installed.
For one auto-migrations typically only work one way and would break rollbacks.

I've added a `package` option to the pict-rs NixOS module that allows users to manually migrate and side step the `stateVersion` based default. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Diff looks good

**Comment by adisbladis** [comment]
@happysalada I inverted the logic in the NixOS module from `is04` to `is03` as I think it's going to be easier to add more backwards compatibility clauses like this. PTAL.

File: pkgs/servers/web-apps/pict-rs/default.nix [hunk 1]
```diff
@@ -13,21 +13,18 @@
...
     domain = "git.asonix.dog";
     owner = "asonix";
     repo = pname;
     rev = "v${version}";
-    sha256 = "mEZBFDR+/aMRFw54Yq+f1gyEz8H+5IggNCpzv3UdDFg=";
+    sha256 = "sha256-1WNd7Ei21g01S5ko6y+uyhHP+xlGtnrwU8MLMxnW3P8=";
   };
 
   cargoLock = {
     lockFile = ./Cargo.lock;
```
Comment (Mic92) [inline pkgs/servers/web-apps/pict-rs/default.nix:27]: You can actually delete the cargo.lock again. Yes, importing from it saves some binary cache storage but it comes at the cost of evaluation memory and nixpkgs tarball size.

Comment (adisbladis) [reply pkgs/servers/web-apps/pict-rs/default.nix]: Done.

**Comment by adisbladis** [comment]
Manually changed tests to use 0.3 and they still work.

**Comment by happysalada** [comment]
all good on my end

**Pull request merged**